detecting and using biome present in system path

This extension should only download the biome binary when it cannot detect it in the system path. This is required for the extension to work on **nixos** because **nixos** is not FHS compliant and it doesn't allow running the biome binary present at `$XDG_DATA_HOME/zed/extensions/work/biome/node_modules/@biomejs/cli-linux-x64/biome` where this extension installs it to by default.

An approach similar to how `markdown-oxide` extension for zed-editor does it would get the job done as shown [here](https://github.com/Feel-ix-343/markdown-oxide-zed/blob/a8789134605237c765eef53855faafcfa8c46e60/src/lib.rs#L22).

For the time being I'm having to provide a custom path to the biome binary in the nix store:
```nix
{
  lib,
  pkgs,
  ...
}: {
  programs.zed-editor = {
    extensions = [
      "biome"
    ];
    enable = true;
    userSettings = {
      lsp = {
        biome = {
          # https://github.com/biomejs/biome-zed/blob/main/CONTRIBUTING.md#custom-biome-binary
          binary = {
            arguments = ["lsp-proxy"];
            path = lib.getExe pkgs.biome;
          };
        };
      };
    };
  };
}
```

I think, this would also be helpful for Docker setups. Running the Zed extension on a macOS host system when working inside a Debian Docker container, you get an error like this in Zed:

```
Language server biome:

initializing server biome, id 11
-- stderr--
node:internal/modules/cjs/loader:1222
  throw err;
  ^

Error: Cannot find module '@biomejs/cli-darwin-arm64/biome'
Require stack:
- /xxx/node_modules/@biomejs/biome/bin/biome
    at Module._resolveFilename (node:internal/modules/cjs/loader:1219:15)
    at Function.resolve (node:internal/modules/helpers:133:19)
    at Object.<anonymous> (/xxx/node_modules/@biomejs/biome/bin/biome:48:11)
    at Module._compile (node:internal/modules/cjs/loader:1504:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1588:10)
    at Module.load (node:internal/modules/cjs/loader:1282:32)
    at Module._load (node:internal/modules/cjs/loader:1098:12)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:215:24)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:158:5) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/xxx/node_modules/@biomejs/biome/bin/biome'
  ]
}

Node.js v22.5.1
```

My current workaround is to install the `@biomejs/cli-darwin-arm64` package in the project (which is not ideal).

--- a/src/biome.rs
+++ b/src/biome.rs
@@ -148,14 +148,6 @@ impl zed::Extension for BiomeExtension {
       });
     }
 
-    if let Some(path) = worktree.which("biome") {
-      return Ok(zed::Command {
-        command: path,
-        args: args,
-        env: Default::default(),
-      });
-    }
-
     // try to run from worktree biome package
     if self.worktree_biome_exists(worktree) {
       let server_path = Path::new(worktree.root_path().as_str())
@@ -169,6 +161,14 @@ impl zed::Extension for BiomeExtension {
       return Ok(zed::Command {
         command: zed::node_binary_path()?,
         args: node_args,
+        env: Default::default(),
+      });
+    }
+
+    if let Some(path) = worktree.which("biome") {
+      return Ok(zed::Command {
+        command: path,
+        args: args,
         env: Default::default(),
       });
     }

--- a/src/biome.rs
+++ b/src/biome.rs
@@ -1,8 +1,9 @@
 use std::path::{Path, PathBuf};
 use zed::settings::LspSettings;
 use zed_extension_api::{
-  self as zed, LanguageServerId, Result, Worktree,
+  self as zed,
   serde_json::{self, Value},
+  LanguageServerId, Result, Worktree,
 };
 
 const WORKTREE_SERVER_PATH: &str = "node_modules/@biomejs/biome/bin/biome";
@@ -140,10 +141,8 @@ impl zed::Extension for BiomeExtension {
     // check and run biome with custom binary
     if let Some(binary) = settings.binary {
       return Ok(zed::Command {
-        command: binary
-          .path
-          .map_or(WORKTREE_SERVER_PATH.to_string(), |path| path),
-        args: binary.arguments.map_or(args, |args| args),
+        command: binary.path.unwrap_or(WORKTREE_SERVER_PATH.to_string()),
+        args: binary.arguments.unwrap_or(args),
         env: Default::default(),
       });
     }

--- a/src/biome.rs
+++ b/src/biome.rs
@@ -148,6 +148,14 @@ impl zed::Extension for BiomeExtension {
       });
     }
 
+    if let Some(path) = worktree.which("biome") {
+      return Ok(zed::Command {
+        command: path,
+        args: args,
+        env: Default::default(),
+      });
+    }
+
     // try to run from worktree biome package
     if self.worktree_biome_exists(worktree) {
       let server_path = Path::new(worktree.root_path().as_str())

--- a/src/biome.rs
+++ b/src/biome.rs
@@ -1,9 +1,8 @@
 use std::path::{Path, PathBuf};
 use zed::settings::LspSettings;
 use zed_extension_api::{
-  self as zed,
+  self as zed, LanguageServerId, Result, Worktree,
   serde_json::{self, Value},
-  LanguageServerId, Result, Worktree,
 };
 
 const WORKTREE_SERVER_PATH: &str = "node_modules/@biomejs/biome/bin/biome";

--- a/src/biome.rs
+++ b/src/biome.rs
@@ -167,7 +167,7 @@ impl zed::Extension for BiomeExtension {
     if let Some(path) = worktree.which("biome") {
       return Ok(zed::Command {
         command: path,
-        args: args,
+        args,
         env: Default::default(),
       });
     }

